docs/path-resolution.rst

```diff
@@ -292,6 +294,14 @@ Here are some examples of what you can expect if they are not:
     import "../.././../util.sol";       // source unit name: util.sol
     import "../../.././../util.sol";    // source unit name: util.sol
 
+.. note::
+
+   This rule may lead to counterintuitive behavior for prefixes that end in relative path segments.
+   For example, when used to compute the import path ``../bar.sol``, the prefix ``a/b/..``
+   (e.g., from a source file ``a/b/../foo.sol``) becomes ``a/b`` and the import path becomes ``bar.sol``.
+   The result is ``a/b/bar.sol``. You may have expected the result to be ``bar.sol`` because
+   ``a/b/../../bar.sol`` evaluates to ``bar.sol`` when normalized as a full path without this rule.
```

Comment:
The examples above already show this case and more.

The initial draft of this page went into detail about this and other unintuitive cases but we intentionally tuned that down not to distract from the main point. I don't think it's a good idea to go back to that version.